src/storage/heap_file.c

```diff
@@ -24863,11 +24851,48 @@ heap_get_visible_version (THREAD_ENTRY * thread_p, const OID * oid, OID * class_
 */
 SCAN_CODE
 heap_scan_get_visible_version (THREAD_ENTRY * thread_p, const OID * oid, OID * class_oid, RECDES * recdes,
-			       HEAP_SCANCACHE * scan_cache, int ispeeking, int old_chn)
+			       RECDES * forward_recdes, HEAP_SCANCACHE * scan_cache, int ispeeking, int old_chn)
 {
   SCAN_CODE scan = S_SUCCESS;
   HEAP_GET_CONTEXT context;
 
+  /*
+   * Before fetching a record, check forward_recdes to see if the record type is REC_HOME,
```

Comment:
This should be done in heap_get_visible_version_internal(), but this part was redundantly written to improve performance. Therefore, I think the need for this code block will be well shared if you clearly state that this is a part of the work you did to improve performance.